### DISCUSSION TITLE
Deleting Single Row

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#2076] [Questions] [answered]
Hi team,

Apologies if I am not submitting this correctly, this is my first post in a discussion. 

I am loving Supabase and wanted to figure out how to delete a single row. In the docs it's using the `match()` filter with a hardcoded id:

```
const { data, error } = await supabase
  .from('cities')
  .delete()
  .match({ id: 666 })
```

Is there a way to link what's being deleted to the primary key ID so it deletes one at a time? If I don't pass anything through it deletes all of them. Here's what I have so far: 

```
    async function deleteData() {
        try {
            const { data, error } = await supabase
            .from('favorite')
            .delete()
            .match()
            // console.log(data[0].id)
        if (error) {
            throw error
        }

        if (data) {
            setdbData(data)
            console.log(data)
        }
        } catch(error) {
            alert(error.message)
        }
    }
    
    return (
        <div>
        {dbData.map((posts, index) => (
            <>
            <a key={index} className='text-blue-500 flex-column' target='_blank' href={posts.url}>{posts.title}</a>
            <button onClick={deleteData} className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded">Delete</button>
            </>
        ))}
        </div>
    )
}
``` 

Thank you for any help!

**Comment by soedirgo** [comment, marked as answer]
Hmm, I don't think there's a way to do this in a single call. You'll have to fetch a single row first, then delete based on that:
```js
const { data: row, error } = await supabase
    .from('favorite')
    .select('id')
    .limit(1)
    .single()
const { data, error } = await supabase
    .from('favorite')
    .delete()
    .eq('id', row.id)
```
Of course, you'll have to handle the case where the row gets modified between the first call and the second call.

**Reply by ghost** [reply]
Thank you for your help, I will give this a go!

**Comment by steve-chavez** [comment]
> Is there a way to link what's being deleted to the primary key ID so it deletes one at a time? If I don't pass anything through it deletes all of them. 

Hey @conor-gaughan,

You mean you want to prevent a single DELETE from deleting all rows right?

This should be possible with the  [safeupdate](https://github.com/eradman/pg-safeupdate) postgres extension. With this extension you wouldn't have to do a `supabase.from().select` first.

Let me know if that would work for you and I can add the extension to your supabase project.
